### PR TITLE
Add more test for custom fakers

### PR DESCRIPTION
**Pull request by snuyanzin**
Currently pass however it might be a guard before PR like https://github.com/datafaker-net/datafaker/pull/1819
is ready to be merged

### CODE REVIEW

File: src/test/java/net/datafaker/providers/base/CustomFakerTest.java [hunk 1]
```diff
@@ -155,4 +156,103 @@ void weightedInsectNameTest() {
...
     }
+
+    public static final class HolderA {
+        public static final class Widget extends AbstractProvider<BaseProviders> {
+            public Widget(BaseProviders faker) {
+                super(faker);
+            }
+
+            @Deterministic
+            public String value() {
```
Comment (snuyanzin) [inline src/test/java/net/datafaker/providers/base/CustomFakerTest.java:167]: edge case for different fakers with same methods

**Review by asolntsev — APPROVED** [review]

**Comment by kingthorin** [comment]
Thanks
